crates/redeem/src/lib.rs

```diff
@@ -336,9 +341,103 @@ pub mod pallet {
             Self::_mint_tokens_for_reimbursed_redeem(vault_id, redeem_id)?;
             Ok(().into())
         }
+
+        #[pallet::weight(<T as Config>::WeightInfo::self_redeem())]
+        #[transactional]
+        pub fn self_redeem(
```

Comment:
Should we bundle this into the existing redeem flow? I guess having a separate extrinsic is cleaner but not sure if there were other considerations in your design here